RedHatInsights/rhsm-subscriptions
# SWATCH-2665: Expose public API by version in different openapi defs

Jira issue: SWATCH-2665

## Description
- Rename existing rhsm-subscriptions-api-spec.yaml to rhsm-subscriptions-api-v1-spec.yaml.
- Generate the v1 resources into the package "org.candlepin.subscriptions.utilization.api.v1"

To ease the review, I splitted the PR into two commits: the actual fix and the move/name of the packages to append the v1. 

## Testing
Verify that the openapi resources are now exposed:
- podman-compose up
- deploy the monolith `DEV_MODE=true ./gradlew :bootRun`
- call the openapi resource: `curl -X GET \
  'http://localhost:8000/api/rhsm-subscriptions/v1/openapi.yaml' \
  -H 'accept: application/vnd.api+json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6IiIsInR5cGUiOiJVc2VyIiwidXNlciI6eyJpc19vcmdfYWRtaW4iOnRydWV9LCJpbnRlcm5hbCI6eyJvcmdfaWQiOiIxMjM0NTYifX19'`

We should now see the expected openapi definition.
